accesscontrol/basic_auth.go

```diff
@@ -0,0 +1,146 @@
+package accesscontrol
+
+import (
+	"bufio"
+	"crypto/subtle"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+)
+
+var _ AccessControl = &BasicAuth{}
+
+var (
+	ErrorBasicAuthNotConfigured = errors.New("basic-auth handler not configured")
+	ErrorBasicAuthUnauthorized  = errors.New("Unauthorized")
+)
+
+const authHeader = "Authorization"
+
+// BasicAuth represents an AC-BasicAuth object
+type BasicAuth struct {
+	htFile htData
+	name   string
+	user   string
+	pass   string
+	realm  string
+}
+
+// NewBasicAuth creates a new AC-BasicAuth object
+func NewBasicAuth(name, user, pass, file, realm string) (*BasicAuth, error) {
```

Comment:
`realm` is currently not used, but should go into the `WWW-Authenticate` response header.